kv/range_cache.go

```diff
@@ -255,12 +255,12 @@ func (rdc *rangeDescriptorCache) clearOverlappingCachedRangeDescriptors(key, met
 	}
 	// Also clear any descriptors which are subsumed by the one we're
 	// going to cache. This could happen on a merge (and also happens
-	// when there's a lot of concurrency). Iterate from StartKey.Next().
+	// when there's a lot of concurrency). Iterate from after StartKey.
```

Comment:
How about:

Iterate from the range meta key after RangeMetaKey(desc.StartKey) to the range meta key for desc.EndKey.